src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs

```
@@ -0,0 +1,117 @@
+        {
+            return Container.GetBlockBlobReference(destinationBlob);
+        }
+
+        public static string CalculateMD5(string filename)
+        {
+            using (var md5 = MD5.Create())
+            {
+                using (var stream = File.OpenRead(filename))
+                {
+                    var hash = md5.ComputeHash(stream);
+                    return Convert.ToBase64String(hash);
+                }
+            }
+        }
+
+        public async Task UploadBlockBlobAsync(string filePath, string blobPath)
+        {
+            CloudBlockBlob cloudBlockBlob = GetBlockBlob(blobPath.Replace("\\", "/"));
+
+            await cloudBlockBlob.UploadFromFileAsync(filePath);
+        }
+
+        public bool IsFileIdenticalToBlob(string blobPath, string localFileFullPath)
+        {
+            CloudBlockBlob blobReference = GetBlockBlob(blobPath);
+
+            return IsFileIdenticalToBlob(localFileFullPath, blobReference);
+        }
+
+        /// <summary>
+        /// Return a bool indicating whether a local file content is same as 
+        /// the content of the pointed blob. If the blob has the ContentMD5
+        /// property set the comparison is performed uniquely using that.
+        /// Otherwise a byte-per-byte comparison with the content of the file
+        /// is performed.
+        /// </summary>
+        public bool IsFileIdenticalToBlob(string localFileFullPath, CloudBlockBlob blobReference)
+        {
+            blobReference.FetchAttributes();
+
+            if (!string.IsNullOrEmpty(blobReference.Properties.ContentMD5))
+            {
+                var localMD5 = CalculateMD5(localFileFullPath);
+                var blobMD5 = blobReference.Properties.ContentMD5;
+
+                return blobMD5.Equals(localMD5, StringComparison.OrdinalIgnoreCase);
+            }
+            else
+            {
+                byte[] existingBytes = new byte[blobReference.Properties.Length];
```

This will fail hard if the file is large, this should use streams to download and compare if the size of the file is too big.

---

The original sources were already doing this comparison so I don't expect anything to fail if the PR is merged. I'll try and use streams to do the comparison, though.

---

@alexperovich PTAL